src/averageAges.js

```diff
@@ -15,11 +15,13 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mens = people.filter(person => person.sex === 'm');
+  const ages = century
+    ? mens.filter(person => Math.ceil(person.died / 100) === century)
+      .map(person => person.died - person.born)
+    : mens.map(person => person.died - person.born);
+
+  return ages.reduce((sum, age) => sum + age, 0) / ages.length;
```

Comment:
Let us move logic of looking for difference, to separate function, to prevent code duplicating